File: backend/src/models/User.model.ts
import mongoose, { Schema, Document } from "mongoose";

export interface IUser extends Document {
  phone?: string;
  email?: string;
  name?: string;
  password: string;
  role: "user" | "admin";
  isPhoneVerified: boolean;
  isDemo: boolean;
  address?: string;
  pincode?: string;
  units?: number;
  orderStatus?: "pending" | "preparing" | "dispatched" | "delivered" | "approved" | "rejected";
  assignedDevices?: string[];
  createdAt: Date;
  updatedAt: Date;
}

const userSchema = new Schema<IUser>(
  {
    phone: {
      type: String,
      unique: true,
      sparse: true,
      trim: true,
    },
    email: {
      type: String,
      unique: true,
      sparse: true,
      lowercase: true,
      trim: true,
    },
    name: {
      type: String,
      trim: true,
    },
    password: {
      type: String,
      required: true,
      minlength: 6,
    },
    role: {
      type: String,
      enum: ["user", "admin"],
      default: "user",
    },
    isPhoneVerified: {
      type: Boolean,
      default: false,
    },
    isDemo: {
      type: Boolean,
      default: false,
    },
    address: {
      type: String,
      trim: true,
    },
    pincode: {
      type: String,
      trim: true,
    },
    units: {
      type: Number,
      default: 1,
    },
    orderStatus: {
      type: String,
      enum: ["pending", "preparing", "dispatched", "delivered", "approved", "rejected"],
      default: "pending",
    },
    assignedDevices: [
      {
        type: String,
        trim: true,
      },
    ],
  },
  { timestamps: true }
);

export const User = mongoose.model<IUser>("User", userSchema);

File: mobile-app/lib/models/user.model.dart
class UserModel {
  final String id;
  final String email;
  final String role;
  final String? name;
  final String? phone;

  UserModel({
    required this.id,
    required this.email,
    required this.role,
    this.name,
    this.phone,
  });

  factory UserModel.fromJson(Map<String, dynamic> json) {
    return UserModel(
      id: json['id'] ?? json['_id'] ?? '',
      email: json['email'] ?? '',
      role: json['role'] ?? 'user',
      name: json['name'],
      phone: json['phone'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'email': email,
      'role': role,
      'name': name,
      'phone': phone,
    };
  }
}
